src/lib/Hydra/Controller/API.pm

```diff
@@ -267,24 +271,76 @@ sub push : Chained('api') PathPart('push') Args(0) {
     );
 }
 
-sub push_github : Chained('api') PathPart('push-github') Args(0) {
+sub webhook_github : Chained('api') PathPart('webhook-github') Args(0) {
     my ($self, $c) = @_;
 
     $c->{stash}->{json}->{jobsetsTriggered} = [];
 
     my $in = $c->request->{data};
-    my $owner = $in->{repository}->{owner}->{name} or die;
-    my $repo = $in->{repository}->{name} or die;
-    print STDERR "got push from GitHub repository $owner/$repo\n";
-
-    triggerJobset($self, $c, $_, 0) foreach $c->model('DB::Jobsets')->search(
-        { 'project.enabled' => 1, 'me.enabled' => 1 },
-        { join => 'project'
-        , where => \ [ 'me.flake like ? or exists (select 1 from JobsetInputAlts where project = me.project and jobset = me.name and value like ?)', [ 'flake', "%github%$owner/$repo%"], [ 'value', "%github.com%$owner/$repo%" ] ]
-        });
+
+    # every GitHub webhook payload has the `repository` key and a `X-GitHub-Event` header
+    my $event = $c->req->header('X-GitHub-Event')        or die;
+    my $owner = (   $in->{repository}->{owner}->{name}
+		 // $in->{repository}->{owner}->{login}) or die;
```

Comment:
You are mixing tabs and spaces quite a lot. Please convert to spaces only.

Reply:
Thanks! That's fixed.